common/src/iosMain/kotlin/com/cuhacking/atlas/common/DataCache.kt

```diff
@@ -0,0 +1,27 @@
+package com.cuhacking.atlas.common
+
+import kotlinx.coroutines.withContext
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import platform.Foundation.*
+
+actual class DataCache {
+    actual var lastModified: Instant? = null
+    private val cacheDirectory = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, true).first()
+    private val file = (cacheDirectory as NSString).stringByAppendingPathComponent("mapdata.json")
+
+    @Suppress("CAST_NEVER_SUCCEEDS")
```

Comment:
What is this for? (just checking)

Reply:
This is for the warning that happens when the Kotlin String is casted to an NSString on line 16. Not sure why it gives a warning. As far as I know, casting a string to an NSString using this syntax is allowed.

Reply:
You're right, it should be fine. Ah well. Probably a bug.